src/app/Systems/_components/SystemToolbar/CreateUserCredentialModal/CreateUserCredentialModal.tsx

```diff
@@ -0,0 +1,269 @@
+import React, { useState, useEffect } from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import * as yup from 'yup';
+import { Systems as Hooks } from '@tapis/tapisui-hooks';
+import { useTapisConfig } from '@tapis/tapisui-hooks';
+import { SystemsApi } from '@tapis/tapis-typescript-systems';
+
+import {
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Button,
+  FormGroup,
+  Label,
+  Input,
+  FormFeedback,
+  Alert,
+} from 'reactstrap';
+
+type CreateUserCredentialModalProps = {
+  toggle: () => void;
+  isOpen: boolean;
+  effectiveUserId: string;
+};
+
+interface FormData {
+  systemId: string;
+  publicKey: string;
+  privateKey: string;
+  loginUser?: string;
+}
+
+// Custom transformation to ensure privateKey is a one-liner
+const transformPrivateKey = (value: string) => value.replace(/\n/gm, '\\n');
+
+const schema = yup.object().shape({
```

Comment:
It would be good to add:
- minimum and maximum character limits. Maximum should be a reasonable number for the field. Minimum is optional.
- you can optionally add in [regex](https://github.com/tapis-project/tapis-ui/blob/eb4ac16e2e61ee6e3fac1e9500d77e7e1e639ca5/src/app/Pods/_components/PodToolbar/CreatePodModal/CreatePodModal.tsx#L193) to give warnings when users inputs invalid characters. (You would need to figure out which characters are in/valid)

[Example 1](https://github.com/tapis-project/tapis-ui/blob/dev/src/app/Apps/_components/AppsToolbar/CreateAppModal/CreateAppModal.tsx)
[Example 2](https://github.com/tapis-project/tapis-ui/blob/dev/src/app/Pods/_components/PodToolbar/CreatePodModal/CreatePodModal.tsx)

Reply:
[
![Screenshot from 2024-07-22 15-25-35](https://github.com/user-attachments/assets/cd212196-6134-446e-9cc0-8d2ff0ed0f23)
](url)

Reply:
Added character limits for the field.

Reply:
![Screenshot from 2024-07-22 15-25-35](https://github.com/user-attachments/assets/8c9bd976-d6ab-41e2-b75f-4ae34d79f42e)